File: Bitwise/869-Reordered_Power_of_2/Solution.py
class Solution:
    def reorderedPowerOf2(self, n: int) -> bool:
        def sorted_string(n):
            return ''.join(sorted(str(n)))
        target = sorted_string(n)
        for i in range(0,31):
            if sorted_string(1 << i) == target:
                return True
        return False

File: Bitwise/869-Reordered_Power_of_2/Solution.js
/**
 * @param {number} n
 * @return {boolean}
 */
var reorderedPowerOf2 = function (n) {
  let target = sortedString(n);
  for (let i = 0; i < 31; i++) {
    if (sortedString(1 << i) == target) {
      return true;
    }
  }
  return false;
};
function sortedString(n) {
  return String(n).split("").sort().join("");
}
